build.sbt

```diff
@@ -140,15 +151,25 @@ lazy val includeGeneratedSrc: Setting[_] = {
 }
 
 lazy val disciplineDependencies = Seq(
-  libraryDependencies ++= Seq("org.scalacheck" %%% "scalacheck" % scalaCheckVersion,
-                              "org.typelevel" %%% "discipline-core" % disciplineVersion
-  )
+  libraryDependencies ++= Seq(
+    "org.scalacheck" %%% "scalacheck" % scalaCheckVersion,
+    "org.typelevel" %%% "discipline-core" % disciplineVersion
+  ).map(_.withDottyCompat(scalaVersion.value))
 )
 
 lazy val testingDependencies = Seq(
   libraryDependencies ++= Seq(
-    "org.typelevel" %%% "discipline-scalatest" % disciplineScalatestVersion % Test,
+    "org.scalatest" %%% "scalatest-shouldmatchers" % scalatestVersion % Test,
+    "org.scalatest" %%% "scalatest-funsuite" % scalatestVersion % Test,
     "org.scalatestplus" %%% "scalacheck-1-14" % scalatestplusScalaCheckVersion % Test
+  ),
+  libraryDependencies ++= Seq(
+    ("org.typelevel" %%% "discipline-scalatest" % disciplineScalatestVersion % Test)
+  ).map(
+    _.exclude("org.scalatestplus", "scalacheck-1-14_2.13")
+      .exclude("org.scalactic", "scalactic_2.13")
+      .exclude("org.scalatest", "scalatest_2.13")
+      .withDottyCompat(scalaVersion.value)
```

Comment:
I'm *tentatively* okay with merging this as-is, but only if we follow up on discipline-scalatest asap to update its dependency. Exclusion rules in published dependencies are dangerous on several levels.

Reply:
What do you think about changing the exclusion to only apply for Dotty (which we’re not currently publishing anyway)? I think it’s likely I’ll be the one publishing discipline-scalatest, and I don’t really want to start doing that for every Dotty release since it’s just a test dependency.

Reply:
@djspiewak Oh, nevermind, I forgot that this is only a test dependency (and that's configured here, so there's no danger of them accidentally missing an `% Test` at the use site), so I don't see any problem with keeping the exclusions as they are now.

Reply:
Ah if it's only a test dependency then I'm okay with it for now.